### ISSUE TITLE
Unhelpful message for tokens after compound commands

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1015]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
if true
then
  while IFS= read -r foo
  do
    true
   done < foo bar
fi
```
or generally just:

```sh
#!/bin/sh
( true ) foo
```
or more realistically:
#### Here's what shellcheck currently says:

Expected 'fi' matching previously mentioned 'if'.

#### Here's what I wanted or expected to see:

Arguments are not valid here. Was this supposed to be on a separate line or part of a redirection?
